src/ui/handlers/starter-select-ui-handler.ts

```diff
@@ -3333,9 +3335,10 @@ export class StarterSelectUiHandler extends MessageUiHandler {
 
     if (species) {
       const dexEntry = globalScene.gameData.dexData[species.speciesId];
-      const abilityAttr = globalScene.gameData.starterData[species.speciesId].abilityAttr;
+      const starterEntry = globalScene.gameData.starterData[species.speciesId];
 
-      const caughtAttr = globalScene.gameData.dexData[species.speciesId]?.caughtAttr || BigInt(0);
+      const abilityAttr = starterEntry.abilityAttr;
+      const caughtAttr = dexEntry?.caughtAttr || BigInt(0);
```

Comment:
Always prefer `??` over `||` unless there is a specific reason to use `||`

```suggestion
      const caughtAttr = dexEntry?.caughtAttr ?? BigInt(0);
```